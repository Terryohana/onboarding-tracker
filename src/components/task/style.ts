import styled from 'styled-components';

export const TaskStyle = styled.div<{isClickable: boolean}>`
  display: flex;
  align-items: center;
  padding: 5px;
  cursor: ${(p) => p.isClickable ? 'pointer' : 'default'};
`;
