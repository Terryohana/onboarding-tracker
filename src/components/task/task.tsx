import React from 'react';
import { useCallback } from 'react';
import { TaskStyle } from './style';

type Props = {
  title: string;
  isCompleted?: false;
  onComplete: () => void;
} | {
  title: string;
  isCompleted: true;
};

export const Task = (props: Props) => {

  const onChange = useCallback(() => props.isCompleted !== true ? props.onComplete() : undefined, [props]);

  return <TaskStyle
    isClickable={!props.isCompleted}
    onClick={onChange}
  >
    <input
      type='checkbox'
      disabled={props.isCompleted}
      checked={!!props.isCompleted}
      onChange={onChange}
    />
    <span>{props.title}</span>
  </TaskStyle>;
}
