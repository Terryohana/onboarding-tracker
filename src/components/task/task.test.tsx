import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, shallow } from 'enzyme';
import { Task } from './task';

configure({ adapter: new Adapter() });

test('Component Task should match snapshot', () => {
  const component = shallow(<Task title='Title' isCompleted={false} onComplete={() => {}}/>);
  expect(component).toMatchSnapshot();
});

test('Component Task has checkbox and title', () => {
  const component = shallow(<Task title='Title' isCompleted={false} onComplete={() => {}}/>);
  expect(component.find('input[type="checkbox"]')).toHaveLength(1);
  expect(component.find('span')).toHaveLength(1);
});

test('Component Task has disabled checkbox if completed', () => {
  const component = shallow(<Task title='Title' isCompleted={true}/>);
  expect(component.find('input[type="checkbox"][disabled]')).toHaveLength(1);
});

test('Component Task calls onComplete', () => {
  const onComplete = jest.fn();
  const component = shallow(<Task title='Title' isCompleted={false} onComplete={onComplete}/>);
  component.simulate('click');
  expect(onComplete.mock.calls.length).toEqual(1);
});
