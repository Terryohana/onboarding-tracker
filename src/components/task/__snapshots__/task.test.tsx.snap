// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Task should match snapshot 1`] = `
<styled.div
  isClickable={true}
  onClick={[Function]}
>
  <input
    checked={false}
    disabled={false}
    onChange={[Function]}
    type="checkbox"
  />
  <span>
    Title
  </span>
</styled.div>
`;
